clientLibrary/src/test/scala/za/co/absa/loginclient/authorization/ClaimsParserTest.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginclient.authorization
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.time.Instant
+import java.util.Base64
+
+class ClaimsParserTest extends AnyFlatSpec with Matchers {
+
+  private val publicKeyString = Base64.getEncoder.encodeToString(FakeTokens.keys.getPublic.getEncoded)
+  private val decoder = JwtDecoderProvider.getDecoderFromPublicKeyString(publicKeyString)
+  private val accessJwt = decoder.decode(FakeTokens.validAccessToken)
+  private val refreshJwt = decoder.decode(FakeTokens.validRefreshToken)
+  private val invalidJwt = decoder.decode(FakeTokens.missingClaimToken)
+
+  "Access token" should "return a subject that equals 'testUser'" in {
+    val subject = AccessTokenClaimsParser.getSubject(accessJwt)
+    assert(subject.equals(FakeTokens.subject))
+  }
+
+  "Access token" should "return a list of groups that contain 'group1', 'group2')" in {
+    val groups = AccessTokenClaimsParser.getGroups(accessJwt)
+    assert(groups == List("group1", "group2"))
+  }
+
+  "Access token" should "return a expiration date that should be within an hour of testing" in {
```

Comment:
```suggestion
 it should "return a expiration date that should be within an hour of testing" in {
```

Using `it` hints connection to the previous test case and creates a nicer structure in IDEA when running tests. Already addressed in #85